test/basic.html

```diff
@@ -217,42 +234,47 @@
         });
 
         test('updates selected item', function(done) {
-          s1.addEventListener('iron-items-changed', function firstListener() {
-            s1.removeEventListener('iron-items-changed', firstListener);
-            var firstElementChild = Polymer.dom(s1).firstElementChild;
+          s1.selected = 0;
+
+          var firstElementChild = Polymer.dom(s1).firstElementChild;
+          expect(firstElementChild).to.be.equal(s1.selectedItem);
+          expect(firstElementChild.classList.contains('iron-selected'))
+              .to.be.eql(true);
+
+          s1.addEventListener('iron-items-changed', function() {
+            firstElementChild = Polymer.dom(s1).firstElementChild;
             expect(firstElementChild).to.be.equal(s1.selectedItem);
             expect(firstElementChild.classList.contains('iron-selected'))
                 .to.be.eql(true);
-            Polymer.dom(s1).removeChild(s1.selectedItem);
-
-            s1.addEventListener('iron-items-changed', function() {
-              firstElementChild = Polymer.dom(s1).firstElementChild;
-              expect(firstElementChild).to.be.equal(s1.selectedItem);
-              expect(firstElementChild.classList.contains('iron-selected'))
-                  .to.be.eql(true);
-              done();
-            });
+            done();
           });
-          s1.selected = 0;
+
+          Polymer.dom(s1).removeChild(s1.selectedItem);
         });
       });
 
       suite('dynamic selector', function() {
+        // NOTE(bicknellr): Polymer 2 only upgrades elements once they have been
+        // connected to the document. This test now connects the selector
+        // *first* and appends the child afterwards.
         test('selects dynamically added child automatically', function(done) {
+          // Create the selector, set selected, connect to force upgrade.
           var selector = document.createElement('iron-selector');
-          var child = document.createElement('div');
-
           selector.selected = '0';
-          child.textContent = 'Item 0';
+          document.body.appendChild(selector);
 
+          // Create and append a new item which should become selected.
+          var child = document.createElement('div');
+          child.textContent = 'Item 0';
           Polymer.dom(selector).appendChild(child);
-          document.body.appendChild(selector);
 
-          Polymer.Base.async(function() {
+          selector.addEventListener('iron-items-changed', function onIronItemsChanged() {
```

Comment:
No, because the `iron-items-changed` event is only fired in the callback for the MutationObserver (`observeNodes`), which doesn't happen synchronously during the `appendChild` above.